src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationService.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationClient;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationRequest;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationResponse;
+import uk.gov.hmcts.reform.notificationservice.data.Notification;
+import uk.gov.hmcts.reform.notificationservice.data.NotificationRepository;
+
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class NotificationService {
+
+    private static final Logger log = getLogger(NotificationService.class);
+
+    private final NotificationRepository notificationRepository;
+    private final ErrorNotificationClient notificationClient;
+
+    public NotificationService(
+        NotificationRepository notificationRepository,
+        ErrorNotificationClient notificationClient
+    ) {
+        this.notificationRepository = notificationRepository;
+        this.notificationClient = notificationClient;
+    }
+
+    public void processPendingNotifications() {
+        List<Notification> notifications = notificationRepository.findPending();
+
+        log.info("Notifications to process: {}", notifications.size());
+
+        notifications
+            .stream()
+            .map(notification -> Pair.of(notification.id, mapToRequest(notification)))
```

Comment:
or you can just pass notification to `processNotifications(...)`

Reply:
only because i need to log all info. right. almost there